rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -411,20 +411,39 @@ impl Controller {
         Ok(())
     }
 
-    fn reload_resource_list(&mut self) -> Result<()> {
-        let Some(session) = &self.session else {
-            tracing::warn!("got notified to update resources but there is no session");
-            return Ok(());
-        };
-        let resources = session.callback_handler.resources.load();
-        let actor_name = &self
-            .auth
-            .session()
-            .ok_or_else(|| anyhow!("connlib is signed in but auth module isn't"))?
-            .actor_name;
-        self.app
-            .tray_handle()
-            .set_menu(system_tray_menu::signed_in(actor_name, &resources))?;
+    /// Returns a new system tray menu
+    fn build_system_tray_menu(&self) -> Result<tauri::SystemTrayMenu> {
+        Ok(if let Some(auth_session) = self.auth.session() {
```

Comment:
I'd rather we have a separate variable that we warp in `Ok`, this looks a bit weird

Reply:
We should just create some helper functions to reduce nesting here, like `signing_in()`, `signed_in()` and `signed_out()` to reduce nesting, it's kind of hard to parse as is.

Reply:
I tried but in the end just left another TODO. 
There are at least 3 different pieces of state to query to figure out whether we're signed in or not:
- Whether a connlib session exists
- Whether we got tunnel_ready, which AFAICT the connlib session doesn't expose directly
- What state the auth state machine is in, since the app as a whole is still signed in even if the user somehow managed to start a second auth flow right as the first one was completing

I can probably refactor the auth state machine later to simplify some of that.